File: GeneratedEntityReaderClasses/Components/ControlsComponent.cs
namespace GeneratedNoitaClasses;
public class ControlsComponent : NoitaComponentBase,  iNoitaType<ControlsComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaBool polymorph_hax { get; set; } = new();
	public NoitaInt polymorph_next_attack_frame { get; set; } = new();
	public NoitaBool enabled { get; set; } = new();
	public NoitaBool gamepad_indirect_aiming_enabled { get; set; } = new();
	public NoitaBool gamepad_fire_on_thumbstick_extend { get; set; } = new();
	public NoitaFloat gamepad_fire_on_thumbstick_extend_threshold { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		polymorph_hax.Read(s);
		polymorph_next_attack_frame.Read(s);
		enabled.Read(s);
		gamepad_indirect_aiming_enabled.Read(s);
		gamepad_fire_on_thumbstick_extend.Read(s);
		gamepad_fire_on_thumbstick_extend_threshold.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		polymorph_hax.Write(s);
		polymorph_next_attack_frame.Write(s);
		enabled.Write(s);
		gamepad_indirect_aiming_enabled.Write(s);
		gamepad_fire_on_thumbstick_extend.Write(s);
		gamepad_fire_on_thumbstick_extend_threshold.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/ControlsComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaInt
from BasicNoitaTypes import NoitaFloat

from NoitaStream import NoitaStream
@dataclass
class ControlsComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    polymorph_hax:NoitaBool = field(default_factory=NoitaBool)
    polymorph_next_attack_frame:NoitaInt = field(default_factory=NoitaInt)
    enabled:NoitaBool = field(default_factory=NoitaBool)
    gamepad_indirect_aiming_enabled:NoitaBool = field(default_factory=NoitaBool)
    gamepad_fire_on_thumbstick_extend:NoitaBool = field(default_factory=NoitaBool)
    gamepad_fire_on_thumbstick_extend_threshold:NoitaFloat = field(default_factory=NoitaFloat)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.polymorph_hax.Read(s)
        self.polymorph_next_attack_frame.Read(s)
        self.enabled.Read(s)
        self.gamepad_indirect_aiming_enabled.Read(s)
        self.gamepad_fire_on_thumbstick_extend.Read(s)
        self.gamepad_fire_on_thumbstick_extend_threshold.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.polymorph_hax.Write(s)
        self.polymorph_next_attack_frame.Write(s)
        self.enabled.Write(s)
        self.gamepad_indirect_aiming_enabled.Write(s)
        self.gamepad_fire_on_thumbstick_extend.Write(s)
        self.gamepad_fire_on_thumbstick_extend_threshold.Write(s)

